src/snowflake/snowpark/mock/snowflake_data_type.py

```diff
@@ -285,6 +286,13 @@ def __init__(self, *args, **kwargs) -> NoReturn:
         sf_type = kwargs.pop("sf_type", None)
         super().__init__(*args, **kwargs)
         self.sf_type: ColumnType = sf_type
+        # record which rows should be marked as null instead of None
+        # snowflake SubfieldString has this behavior
+        # suppose there are two Variant objects in table "v": 1. { "a": None } 2. None
```

Comment:
I understand this is to align with live connection behavior. But I think this is a buggy behavior, let's try to fix it main and revert this in local testing.

Reply:
JIRA created: SNOW-960190